# Walidacja: zestawy cech na nowych danych (Sprint 4+)

## Cel
Spojna, swieza walidacja walk-forward wszystkich zestawow cech na nowych danych: surface_speed,
fatigue, enriched (surface+fatigue), Elo. Per sezon: baseline (cache) + kazdy zestaw na tych samych
meczach, parowanie + McNemar.

In [1]:
import sys
from pathlib import Path
sys.path.insert(0, str(Path("../src").resolve()))

## Uruchomienie (od zera)

In [2]:
import tennis_model_validate_features as m
m.main()


===== ROK 2020 =====


  baseline match=0.6176


    surface    match=0.6250  delta=+0.0074


    fatigue    match=0.6140  delta=-0.0037


    enriched   match=0.6140  delta=-0.0037


    elo        match=0.6434  delta=+0.0257



===== ROK 2021 =====


  baseline match=0.6667


    surface    match=0.6762  delta=+0.0096


    fatigue    match=0.6609  delta=-0.0057


    enriched   match=0.6667  delta=+0.0000


    elo        match=0.6743  delta=+0.0077



===== ROK 2022 =====


  baseline match=0.6728


    surface    match=0.6782  delta=+0.0055


    fatigue    match=0.6746  delta=+0.0018


    enriched   match=0.6764  delta=+0.0037


    elo        match=0.6965  delta=+0.0238



===== ROK 2023 =====


  baseline match=0.6346


    surface    match=0.6435  delta=+0.0089


    fatigue    match=0.6381  delta=+0.0036


    enriched   match=0.6346  delta=+0.0000


    elo        match=0.6328  delta=-0.0018



===== ROK 2024 =====


  baseline match=0.6186


    surface    match=0.6237  delta=+0.0051


    fatigue    match=0.6220  delta=+0.0034


    enriched   match=0.6254  delta=+0.0068


    elo        match=0.6356  delta=+0.0169



===== ROK 2025 =====


  baseline match=0.6566


    surface    match=0.6566  delta=+0.0000


    fatigue    match=0.6566  delta=+0.0000


    enriched   match=0.6585  delta=+0.0019


    elo        match=0.6377  delta=-0.0189



WALK-FORWARD CECH (nowe dane 2020-2025) vs baseline

--- surface (3 cech) ---
 year  baseline  variant  delta
 2020    0.6176   0.6250 0.0074
 2021    0.6667   0.6762 0.0096
 2022    0.6728   0.6782 0.0055
 2023    0.6346   0.6435 0.0089
 2024    0.6186   0.6237 0.0051
 2025    0.6566   0.6566 0.0000
  POOLED (3022): baseline=0.6463  surface=0.6522  delta=+0.0060  (dodatnie 5/6 lat)
  McNemar: b=46 c=64 z=1.62 p=0.1050 -> brak istotnosci

--- fatigue (6 cech) ---
 year  baseline  variant   delta
 2020    0.6176   0.6140 -0.0037
 2021    0.6667   0.6609 -0.0057
 2022    0.6728   0.6746  0.0018
 2023    0.6346   0.6381  0.0036
 2024    0.6186   0.6220  0.0034
 2025    0.6566   0.6566  0.0000
  POOLED (3022): baseline=0.6463  fatigue=0.6466  delta=+0.0003  (dodatnie 3/6 lat)
  McNemar: b=54 c=55 z=0.00 p=1.0000 -> brak istotnosci

--- enriched (9 cech) ---
 year  baseline  variant   delta
 2020    0.6176   0.6140 -0.0037
 2021    0.6667   0.6667  0.0000
 2022    0.6728   0.6764  0.0037
 

## Wnioski
Na walk-forward (2020-2025, N=3022) ZADEN zestaw cech nie jest istotny statystycznie -- wszystkie
McNemar p > 0.05 (najnizsze surface p=0.105, elo p=0.173). Pooled delty sa kierunkowo dodatnie, ale
w granicach szumu: elo +0.76, surface +0.60, enriched +0.20, fatigue +0.03 p.p. To komplet dowodow,
ze nic nie przebija baseline -- sufit ~65% jest odporny na nowe cechy.